core/src/player.rs

```diff
@@ -556,6 +556,99 @@ impl Player {
         self.is_playing
     }
 
+    fn get_allowed_context_menu_items<'gc>(
+        context: &mut UpdateContext<'_, 'gc, '_>,
+    ) -> Vec<&'static str> {
+        let mut activation = Activation::from_stub(
+            context.reborrow(),
+            ActivationIdentifier::root("[ContextMenu]"),
+        );
+
+        let dobj = activation.context.stage.root_clip();
+        let is_multiframe_movie = dobj
+            .as_movie_clip()
+            .map(|mc| mc.total_frames() > 1)
+            .unwrap_or(false);
+        let mut names = if is_multiframe_movie {
+            vec![
+                "zoom",
+                "quality",
+                "play",
+                "loop",
+                "rewind",
+                "forward_back",
+                "print",
+            ]
+        } else {
+            vec!["zoom", "quality", "print"]
+        };
+
+        if let Value::Object(obj) = dobj.object() {
+            if let Ok(Value::Object(menu)) = obj.get("menu", &mut activation) {
+                if let Ok(Value::Object(menu)) = menu.get("builtInItems", &mut activation) {
+                    names.retain(|name| {
+                        !matches!(menu.get(name, &mut activation), Ok(Value::Bool(false)))
+                    });
+                }
+            }
+        }
+        names
+    }
+
+    pub fn get_builtin_menu_items(&mut self) -> Vec<&'static str> {
+        self.mutate_with_update_context(|context| {
+            if !context.stage.show_menu() {
+                return vec![];
+            }
+            Self::get_allowed_context_menu_items(context)
```

Comment:
Let's inline this here instead of having the extra fn.

Reply:
My reasoning was that when AVM2 ContextMenu support gets added, it'd require a similar-but-slightly-different chunk of code, which most likely would be refactored into a separate function anyway. But yeah, it can be inlined for now.